pkg/autoscaler/stats_scraper.go

```diff
@@ -142,22 +144,48 @@ func (s *ServiceScraper) Scrape() (*StatMessage, error) {
 	sampleSize := populationMeanSampleSize(readyPodsCount)
 	statCh := make(chan *Stat, sampleSize)
 
-	url := s.target()
+	scrapedPods := sets.NewString()
+	var scrapedPodMux sync.Mutex
+	writeIfAbsent := func(name string) bool {
```

Comment:
it's more `checkIfAbsent`.

Reply:
Hmm, I disagree here. Renamed to `addIfAbsent` since it also adds the key to the set if it's not already there.